File: 0049-group-anagrams/0049-group-anagrams.go
func groupAnagrams(strs []string) [][]string {
    m := make(map[string][]string)
    
    for _ , word := range strs {
        chars := []rune(word)
        sort.Slice(chars, func(i, j int) bool {
            return chars[i] < chars[j]
         })
    
        m[string(chars)] = append(m[string(chars)], word)
    }
    
    var result [][]string
    for _, str := range m {
        result = append(result, str)
    }
    
    return result
    
    
    
      
   
    
}

File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    
    string hash_code(string word){
        string code = "";
        int arr[26] = {0};
        
        for(char c : word){
            arr[c -'a']++;
        }
        
        int freq;
        for(int i = 0; i < 26; i++){
            freq = arr[i];
            if(freq>0){
                code+=string(freq, i + 'a');
            }
        }
        
        return code;
    }
    
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map< string, vector<string>> m;
        
        for(string word: strs){
            m[hash_code(word)].push_back(word);
        }
        
        vector<vector<string>> result;
        
        for(pair< string, vector<string>> str: m){
            result.push_back(str.second);
        }
        
        return result;
    }
};